src/lib/comms/sol-socket-impl-contiki.c

```diff
@@ -0,0 +1,274 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+
+#include <contiki-net.h>
+
+#include "sol-event-handler-contiki.h"
+#include "sol-log.h"
+#include "sol-mainloop.h"
+#include "sol-socket-impl.h"
+#include "sol-network-util.h"
+#include "sol-vector.h"
+
+struct sol_socket_contiki {
+    struct sol_socket base;
+
+    bool (*on_can_read)(void *data, struct sol_socket *s);
+    bool (*on_can_write)(void *data, struct sol_socket *s);
+    const void *data;
+
+    struct simple_udp_connection udpconn;
+
+    struct sol_ptr_vector pending_buffers;
+    struct sol_timeout *write_timeout;
+
+    bool read_monitor;
+    bool write_monitor;
+};
+
+struct pending_buffer {
+    struct sol_network_link_addr addr;
+    uint16_t datalen;
+    uint8_t data[];
+};
+
+extern struct process soletta_app_process;
+process_event_t sol_udp_socket_event;
+static bool registered_process = false;
+
+#define container_of(ptr, type, member) \
```

Comment:
On Mon, 18 Apr 2016 11:26:55 -0700, Vinicius Gomes wrote:
> > +
> > +    bool read_monitor;
> > +    bool write_monitor;
> > +};
> > +
> > +struct pending_buffer {
> > +    struct sol_network_link_addr addr;
> > +    uint16_t datalen;
> > +    uint8_t data[];
> > +};
> > +
> > +extern struct process soletta_app_process;
> > +process_event_t sol_udp_socket_event;
> > +static bool registered_process = false;
> > +
> > +#define container_of(ptr, type, member) \
> 
> Would this be generally useful, so it can be promoted to the `sol_` namespace?

I don't know, anyone has a use for it now?